app/(tabs)/movie/[id].tsx

```diff
@@ -200,62 +204,124 @@ export default function MovieDetailsScreen() {
+            [actorWithCast.firstName, actorWithCast.lastName]
+              .filter(Boolean)
+              .join(' ')
+              .trim() ||
+            actorWithCast.nickName ||
+            ''
+
+          if (!name) {
+            return null
+          }
+
+          const id = actorWithCast.id ?? String(actorWithCast.tmdb_id ?? name)
+
+          // Get character info from nested cast object
+          const castInfo = actorWithCast.cast
+          const character =
+            (Array.isArray(castInfo?.characters) &&
+            castInfo.characters.length > 0
+              ? castInfo.characters.join(', ')
+              : null) ||
+            castInfo?.role ||
+            undefined
+
+          const image =
+            actorWithCast.profile_image &&
+            actorWithCast.profile_image.trim().length > 0
+              ? actorWithCast.profile_image
+              : undefined
+
+          const normalized: NormalizedCastMember = {
+            id: String(id),
+            name,
+          }
+
+          if (character) {
+            normalized.character = character
+          }
+
+          if (image) {
+            normalized.image = image
+          }
+
+          return normalized
+        })
+        .filter((member): member is NormalizedCastMember => member !== null)
+    }
+
+    // Fallback to legacy structure: movie.cast (array of cast entries with actor nested)
+    if (Array.isArray(movie.cast) && movie.cast.length > 0) {
+      return movie.cast
+        .map((member: any) => {
+          if (!member) {
+            return null
+          }
+
+          const actor = member.actor
+          if (!actor) {
+            return null
+          }
+
+          const name =
+            [actor.firstName, actor.lastName]
+              .filter(Boolean)
+              .join(' ')
+              .trim() ||
+            actor.nickName ||
+            ''
+
+          if (!name) {
+            return null
+          }
+
+          const id =
+            actor.id ??
+            member.actor_id ??
+            (member.movie_id ? `${member.movie_id}-${name}` : name)
+
+          const character =
+            (Array.isArray(member.characters) && member.characters.length > 0
+              ? member.characters.join(', ')
+              : null) ||
+            member.role ||
+            undefined
+
+          const image =
+            actor.profile_image && actor.profile_image.trim().length > 0
+              ? actor.profile_image
+              : undefined
+
+          const normalized: NormalizedCastMember = {
+            id: String(id),
+            name,
+          }
+
+          if (character) {
+            normalized.character = character
+          }
+
+          if (image) {
+            normalized.image = image
+          }
+
+          return normalized
+        })
+        .filter((member): member is NormalizedCastMember => member !== null)
+    }
+
+    return []
+  }, [movie.actors, movie.cast])
```

Comment:
Why is this change necessary? Please check what does the `get /movies/:id` endpoint returns and only use the field that the endpoint returns. The other fallback is not necessary we rely in the endpoint as the only source of truth